File: src/022/p022.py
import os.path


def get_name_value(name: str) -> int:
    return sum(ord(char) - ord("A") + 1 for char in name)


def get_name_score(name: str, index: int) -> int:
    return get_name_value(name) * (index + 1)


def solve() -> int:
    input_path = os.path.join(os.path.dirname(__file__), "p022_names.txt")
    with open(input_path, encoding="utf8") as file:
        content = file.read()
    names = sorted(name.strip('"') for name in content.split(","))
    return sum(get_name_score(name, index) for index, name in enumerate(names))

File: src/022/p022.go
package problem022

import (
	"embed"
	"sort"
	"strconv"
	"strings"

	"github.com/sunilbpandey/project-euler/src/utils/go/errorutils"
)

//go:embed p022_names.txt
var content embed.FS

func getNameValue(name string) int {
	value := 0
	for _, c := range name {
		value += int(c) - 'A' + 1
	}
	return value
}

func Solve() string {
	content, err := content.ReadFile("p022_names.txt")
	errorutils.Check(err)

	score := 0
	names := strings.Split(string(content), ",")
	sort.Strings(names)
	for i, name := range names {
		name = strings.Trim(name, "\"")
		score += getNameValue(name) * (i + 1)
	}
	return strconv.Itoa(score)
}
